modules/twinkledeprod.js

```diff
@@ -48,38 +48,43 @@ Twinkle.deprod.callback = function() {
 		'gcmnamespace': '0|6|108|2', // mostly to ignore categories
 		'prop': [ 'info', 'revisions' ],
 		'rvprop': [ 'content' ],
-		'inprop': [ 'protection' ]
+		'inprop': [ 'protection' ],
+		'format': 'json'
 	};
 
 	var statelem = new Morebits.status('Grabbing list of pages');
 	var wikipedia_api = new Morebits.wiki.api('loading...', query, function(apiobj) {
-		var $doc = $(apiobj.responseXML);
-		var $pages = $doc.find('page[ns!="6"]');  // all non-files
+		var response = apiobj.getResponse();
+		var pages = response.query.pages.filter(function(p) {
```

Comment:
Breaks when invoked on an empty category

Reply:
Indeed!  Handled in batchdelete and batchprotect, but not here or batchundelete, so fixed x2.